src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -196,9 +218,9 @@ public PinValue Read(int pinNumber)
         /// </summary>
         /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
         /// <param name="value">The value to be written to the pin.</param>
-        public void Write(int pinNumber, PinValue value)
+        public virtual void Write(int pinNumber, PinValue value)
```

Comment:
I'm curious, why all these need to be virtual instead of replacing the driver? the initial idea was that you should be changing the driver, not the controller (although maybe would be simpler this way if we did that from the start)

Reply:
@joperezr and me came up with the suggestion to unseal the GpioController because of several issues:

- If you want to add features to the GpioController (such as pin management, for the board approach), there's currently no way to do that (because GpioController is sealed). 
- Since all methods (except the .ctor) on the Driver class are protected, it is not even possible to create a replacement for the GpioController and call into existing driver implementations.
- A sealed class cannot be mocked for unit testing

Therefore, the controller is now unsealed and all methods that you might want to override or mock are virtual. The change should not be breaking for anyone.

Reply:
fine with me

Reply:
cc: @joperezr

Reply:
This has been done now